driver/src/main/java/org/neo4j/driver/internal/net/pooling/SocketConnectionPool.java

```diff
@@ -106,34 +124,57 @@ private BlockingPooledConnectionQueue pool( BoltServerAddress address )
         return pool;
     }
 
-    @Override
-    public void purge( BoltServerAddress address )
+    private PooledConnection acquireConnection( final BoltServerAddress address,
+            final BlockingPooledConnectionQueue connectionQueue )
     {
-        BlockingPooledConnectionQueue connections = pools.remove( address );
-        if ( connections != null )
+        Supplier<PooledConnection> supplier = newPooledConnectionSupplier( address, connectionQueue );
+
+        int acquisitionAttempt = 1;
+        PooledConnection connection = connectionQueue.acquire( supplier );
+        while ( !canBeAcquired( connection, acquisitionAttempt ) )
```

Comment:
Changed code to track when connection was created and then return it. So it'll retry until new connection is created and then return it to the user.